app/controllers/carts_controller.rb

```diff
@@ -0,0 +1,18 @@
+class CartsController < ApplicationController
+  def add
+    dish = Dish.find(params[:id])
+    if Cart.any?
+      find_cart
```

Comment:
If Cart.any? Means if there's any carts in the db. What you should use here is a check if the session[:cart_id] is set. If it is, get that Cart (as you do with the 'find_cart' method). Also,  I would suggest saving the cart id to the session in the 'else' block not outside. That only has to happen once per session. If you are unsure about sessions in Rails, pls check Rails docs.